frame/assets/src/functions.rs

```diff
@@ -345,7 +353,10 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			account.reason.take_deposit(&who).ok_or(Error::<T, I>::NoDeposit)?;
 		let mut details = Asset::<T, I>::get(&id).ok_or(Error::<T, I>::Unknown)?;
 		ensure!(details.status == AssetStatus::Live, Error::<T, I>::AssetNotLive);
-		ensure!(account.balance.is_zero() || allow_burn, Error::<T, I>::WouldBurn);
+		ensure!(
+			account.balance.is_zero() || allow_burn || &depositor != who,
```

Comment:
This isn't safe. A foreign depositor should only be able to refund their deposit if the balance is zero. Otherwise a foreign deposit could be placed on the account, funds transferred in and then the funds lost when the foreign depositor pulls the rug.

Reply:
removed, not possible to refund_other for non zero balance now